packages/parameters/tests/e2e/ssmProvider.class.test.functionCode.ts

```diff
@@ -0,0 +1,183 @@
+const customClient = new SSMClient({});
+customClient.middlewareStack.use(middleware);
+const providerWithMiddleware = new SSMProvider({
+  awsSdkV3Client: customClient
+});
+
+const paramA = process.env.PARAM_A ?? 'my-param';
+const paramB = process.env.PARAM_B ?? 'my-param';
+const paramEncryptedA = process.env.PARAM_ENCRYPTED_A ?? 'my-encrypted-param';
+const paramEncryptedB = process.env.PARAM_ENCRYPTED_B ?? 'my-encrypted-param';
+
+// Use provider specified, or default to main one & return it with cache cleared
+const resolveProvider = (provider?: SSMProvider): SSMProvider => {
+  const resolvedProvider = provider ? provider : defaultProvider;
+  resolvedProvider.clearCache();
+
+  return resolvedProvider;
+};
+
+// Helper function to call get() and log the result
+const _call_get = async (
+  paramName: string,
+  testName: string,
+  options?: SSMGetOptionsInterface,
+  provider?: SSMProvider
+): Promise<void> => {
+  try {
+    const currentProvider = resolveProvider(provider);
+
+    const parameterValue = await currentProvider.get(paramName, options);
+    logger.log({
+      test: testName,
+      value: parameterValue
+    });
+  } catch (err) {
+    logger.log({
+      test: testName,
+      error: err.message
+    });
+  }
+};
+
+// Helper function to call getMultiple() and log the result
+const _call_get_multiple = async (
+  paramPath: string,
+  testName: string,
+  options?: SSMGetMultipleOptionsInterface,
+  provider?: SSMProvider
+): Promise<void> => {
+  try {
+    const currentProvider = resolveProvider(provider);
+
+    const parameterValues = await currentProvider.getMultiple(
+      paramPath,
+      options
+    );
+    logger.log({
+      test: testName,
+      value: parameterValues
+    });
+  } catch (err) {
+    logger.log({
+      test: testName,
+      error: err.message
+    });
+  }
+};
+
+// Helper function to call getParametersByName() and log the result
+const _call_get_parameters_by_name = async (
+  params: Record<string, SSMGetParametersByNameOptionsInterface>,
+  testName: string,
+  options?: SSMGetParametersByNameOptionsInterface,
+  provider?: SSMProvider
+): Promise<void> => {
+  try {
+    const currentProvider = resolveProvider(provider);
+
+    const parameterValues = await currentProvider.getParametersByName(params, options);
+    logger.log({
+      test: testName,
+      value: parameterValues
+    });
+  } catch (err) {
+    logger.log({
+      test: testName,
+      error: err.message
+    });
+  }
+};
+
+export const handler = async (_event: unknown, _context: Context): Promise<void> => {
+  // Test 1 - get a single parameter by name with default options
+  await _call_get(paramA, 'get');
+  
+  // Test 2 - get a single parameter by name with decrypt
+  await _call_get(paramEncryptedA, 'get-decrypt', { decrypt: true });
+  
+  // Test 3 - get multiple parameters by path with default options
+  // Get path (/param/get)
+  const parameterPath = paramA.substring(0, paramA.lastIndexOf('/'));
```

Comment:
I see that you want to recycle params in different tests. While the comment above helps to understand the value, it's still difficult to reason, because we use one parameter in different test cases.

I'd suggest so separate the resources and dedicate one specific set of parameter for each test case. This will prevent any unexpected side effects, if maintainers will change `paramA` in the future, for the first test and might break other tests.

Reply:
Yes, I was undecided about this as I agree it's a good practice for tests to be independent.

However here I was trying to balance this best practice, with the fact that I would want to avoid the tests running forever due to having to create and destroy many resources at each run. 

Thoughts on this?

Reply:
I see your point. I tried to create params per test case it bloats the setup excessively. I am not concerned with the execution time, the results in my tests were very similar, between 165 and 175 seconds. 

Following back to tenets, we should keep it simple like it is right now.